src/style-spec/read_style.js

```diff
@@ -0,0 +1,15 @@
+import ParsingError from './error/parsing_error';
+import jsonlint from '@mapbox/jsonlint-lines-primitives';
+
+export default function readStyle(style) {
+    let s = style;
+    if (s instanceof String || typeof s === 'string' || s instanceof Buffer) {
+        try {
+            s = jsonlint.parse(s.toString());
```

Comment:
Could we return here instead of the new variable assignment?

```suggestion
            return jsonlint.parse(style.toString());
```

Then we could also remove the `let s = style;` and change the final return to `return style`